crates/oxc_linter/src/rules/jsx_a11y/no_redundant_roles.rs

```diff
@@ -0,0 +1,122 @@
+use crate::{
+    context::LintContext,
+    rule::Rule,
+    utils::{get_element_type, has_jsx_prop_lowercase},
+    AstNode,
+};
+use oxc_ast::{
+    ast::{JSXAttributeItem, JSXAttributeValue},
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use phf::{phf_map, phf_set};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error(
+    "eslint-plugin-jsx-a11y(no-redundant-roles): The element `{element}` has an implicit role of `{role}`. Defining this explicitly is redundant and should be avoided."
+)]
+#[diagnostic(
+    severity(warning),
+    help("Remove the redundant role `{role}` from the element `{element}`.")
+)]
+struct NoRedundantRolesDiagnostic {
+    #[label]
+    pub span: Span,
+    pub element: String,
+    pub role: String,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoRedundantRoles;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Enforces that the explicit role property is not the same as implicit/default role property on element.
+    ///
+    /// ### Why is this bad?
+    /// Redundant roles can lead to confusion and verbosity in the codebase.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// <nav role="navigation" />
+    ///
+    /// // Good
+    /// <nav />
+    /// ```
+    NoRedundantRoles,
+    correctness
+);
+
+static DEFAULT_ROLE_EXCEPTIONS: phf::Map<&'static str, phf::Set<&'static str>> = phf_map! {
+    "nav" => phf_set!{"navigation"},
```

Comment:
one last bit.

does this need to be a `Map<str, Set<str>>`. couldn't it just be `Map<str, str>` since all of the value sets only have one element

Reply:
Thanks for the review!
You're right, `Map<str, str>` looked good to me: 🙇‍♂️ 36f6108